object_store/src/lib.rs

```diff
@@ -589,15 +640,22 @@ impl Error {
         match self.0 {
             UnableToPutDataToS3 {
                 source: RusotoError::Credentials(_),
+                bucket: String(_),
+                location: String(_),
```

Comment:
I think these should be:

```suggestion
                bucket: _,
                location: _,
```

As the `String` is not a tuple:

```
error[E0532]: expected tuple struct or tuple variant, found struct `String`
   --> object_store/src/lib.rs:643:25
    |
643 |                 bucket: String(_),
    |                         ^^^^^^^^^ help: use struct pattern syntax instead: `String { vec }`
    |
   ::: /Users/dom/.rustup/toolchains/nightly-2020-11-19-x86_64-apple-darwin/lib/rustlib/src/rust/library/alloc/src/string.rs:278:1
    |
278 | pub struct String {
    | ----------------- `String` defined here
```

I had to change it to this so it would compile for me. We should be checking that the code compiles with these `cfg`'s even if we're not currently running the tests, so I've made #537 to track this.

Reply:
Updated `String(_)` to `_` in the pattern match